### ISSUE TITLE
Flux.train error when using reduce hcat in loss

### ISSUE THREAD

**Issue by swone** [FluxML/Zygote.jl#626]
```julia
using Flux #, Optim

function loss(x, y)
      _pred = zeros((20, 56))
      pred = reduce(hcat,[_pred[:,i] for i in 1:size(_pred,2)])
      loss = sum(abs2, pred .- y)
      loss
end

v0 = zeros(15)
sol_data = zeros((20, 56))
display(loss(v0, sol_data))
dataset = [(v0, sol_data)]
p = 0
Flux.train!(loss, Flux.params(p), dataset, ADAM())
```
The loss function runs properly when it is separately called, but when using the loss function in Flux.train gives a `Mutating arrays is not supported` error.

**Comment by AzamatB** [comment]
Can you try if  #501 fixes this?

**Comment by mcabbott** [comment]
You may also like [SliceMap](https://github.com/mcabbott/SliceMap.jl) for this, which besides a gradient for `reduce(hcat` should be more efficient than doing each `_pred[:,i]` independently.

**Comment by swone** [comment]
@AzamatB How do I test #501 locally?

**Comment by swone** [comment]
@AzamatB How do I try it out?

**Comment by dhairyagandhi96** [comment]
You might find this helpful https://stackoverflow.com/questions/27567846/how-can-i-check-out-a-github-pull-request-with-git

**Comment by swone** [comment]
I have the pull request on my computer now, but I'm not sure how to test the code using it.

**Comment by swone** [comment]
@AzamatB I am still getting the same error when I try it with #501

**Comment by swone** [comment]
> Can you try if #501 fixes this?

This fixes it!
